staging/src/k8s.io/api/extensions/v1beta1/register.go

```diff
@@ -28,6 +28,11 @@ const GroupName = "extensions"
 // SchemeGroupVersion is group version used to register these objects
 var SchemeGroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1beta1"}
 
+// Kind takes an unqualified kind and returns a Group qualified GroupKind
+func Kind(kind string) schema.GroupKind {
```

Comment:
Lets not introduce this now, and instead just put it where it is used.  We want to solve this a different way anyway.

Reply:
Do you mean I should define the function in all the files that use it, such as rollout_status.go and history.go?

Reply:
The func is version independent. That's why we only have it in the internal types. Just use `SchemeGroupVersion.WithKind(kind).GroupKind()` where it is used. After that lgtm.